crepecake/src/main/java/com/google/cloud/tools/crepecake/builder/Step.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright 2018 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.crepecake.builder;
+
+/** A step in the builder process. Implementations must be thread-safe. */
+@FunctionalInterface
+interface Step<T, R> {
+
+  R run(T input) throws Exception;
```

Comment:
probably a bad idea for functional methods to throw exceptions; and we really shouldn't throw an arbitrary exception